lib/install.js

```diff
@@ -859,23 +859,29 @@ function visitForkRanges(tree, resolution, name, parentName, visit) {
   Object.keys(tree.depMap).forEach(function(parent) {
     var curDepMap = tree.depMap[parent];
 
-    Object.keys(curDepMap).forEach(function(dep) {
-      var secondary = curDepMap[dep];
+    Object.keys(curDepMap)
+      .filter(function(dep) {
+        return matchesSecondaryDepName(dep) && isNotForkOfSelf(dep) && hasRange(dep);
+      })
+      .forEach(function(dep) {
+        var ranges = secondaryRanges[parent];
+        var deps = ranges && ranges.deps && ranges.deps[dep];
+        var secondary = curDepMap[dep];
 
-      if (secondary.name !== resolution.name)
-        return;
+        visit(dep, parent, secondary, deps);
+      });
 
-      // its not a fork of itself
-      if (dep === name && parent === parentName)
-        return;
 
-      // skip if we don't have a range
+    function matchesSecondaryDepName(dep) {
+      return curDepMap[dep].name === resolution.name;
+    }
+    function isNotForkOfSelf(dep) {
+      return dep !== name || parent !== parentName;
+    }
+    function hasRange(dep) {
       var ranges = secondaryRanges[parent];
-      if (!ranges || !ranges[dep])
```

Comment:
Would the simple fix not be to just adjust this to `!ranges.deps[dep]`?

I know your adjustment is easier to read, but involves unnecessary repeated dictionary lookups, so in the name of performance I think it may be best to keep this unrolled.